tox.ini

```diff
@@ -6,8 +6,32 @@ envlist = py37,py38,py39,py310,lint
 whitelist_externals = poetry
 commands =
     poetry install -v -E testing
-    poetry run pytest .
+    poetry run pytest {posargs:.}
+
+[testenv:all-checks]
+deps = pre-commit
+commands = pre-commit run --color=always {posargs:--all-files}
 
 [testenv:lint]
 deps = pre-commit
-commands = pre-commit run --all-files --show-diff-on-failure
+commands = pre-commit run pylint --color=always {posargs:--all-files}
+
+[testenv:fmt]
+deps = pre-commit
+ignore_errors = true
+commands =
+    pre-commit run black --color=always {posargs:--all-files}
+    pre-commit run isort --color=always {posargs:--all-files}
+    pre-commit run prettier --color=always {posargs:--all-files}
+    pre-commit run blacken-docs --color=always {posargs:--all-files}
```

Comment:
Are there more pre-commit hooks that modify files (rather than just display the error/warning)?

Reply:
The doc string one does.

Reply:
I assume you mean `docformatter`?

I also added `end-of-file-fixer` and `trailing-whitespace`